### PR TITLE
SDIT-1593: ✨ More switching to authentication holder

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public List<OffenderSentenceDetail> getOffenderSentences(@RequestParam(value = "
     @GetMapping("/home-detention-curfew-candidates")
     @SlowReportQuery
     public List<OffenderSentenceCalc<BaseSentenceCalcDates>> getOffenderSentencesHomeDetentionCurfewCandidates() {
-        return offenderCurfewService.getHomeDetentionCurfewCandidates(authenticationFacade.getCurrentPrincipal());
+        return offenderCurfewService.getHomeDetentionCurfewCandidates(hmppsAuthenticationHolder.getUsername());
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java:95]: this always expects a `username` anyway

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java [hunk 2]
```diff
@@ -194,7 +193,7 @@ public List<Agency> findAgenciesByUsername(final String username, final boolean
      * @return set of agency location ids accessible to current authenticated user.
      */
     public Set<String> getAgencyIds(boolean allowInactive) {
-        return findAgenciesByUsername(authenticationFacade.getCurrentPrincipal(), allowInactive)
+        return findAgenciesByUsername(hmppsAuthenticationHolder.getUsername(), allowInactive)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:196]: `findAgencies` always expects a `username`

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/StaffAwareMovementService.kt [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
 ) {
-  internal fun getLoggedInStaff(): Result<StaffUserAccount> {
-    return staffUserAccountRepository.findByIdOrNull(authenticationFacade.currentPrincipal)
-      ?.let { Result.success(it) } ?: Result.failure(
-      EntityNotFoundException.withId(authenticationFacade.currentPrincipal),
-    )
-  }
+  internal fun getLoggedInStaff(): Result<StaffUserAccount> =
+    hmppsAuthenticationHolder.username?. let {
+      staffUserAccountRepository.findByIdOrNull(hmppsAuthenticationHolder.username)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/StaffAwareMovementService.kt:15]: this always needs a `username`

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/StaffAwareMovementService.kt [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
+        ?.let { Result.success(it) } ?: Result.failure(
+        EntityNotFoundException.withId(hmppsAuthenticationHolder.username),
+      )
+    } ?: Result.failure(EntityNotFoundException.withId("no username supplied"))
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/StaffAwareMovementService.kt:19]: had to do a `null` check here though to error if `username` not supplied

File: src/test/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/CaseNoteMovementServiceTest.kt [hunk 5]
```diff
@@ -142,6 +143,20 @@ internal class CaseNoteMovementServiceTest {
         service.createGenerateAdmissionNote(booking, movement)
       }
     }
+
+    @Test
+    internal fun `will throw exception if authentication doesn't contain a user`() {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/CaseNoteMovementServiceTest.kt:148]: added more tests for my new null check code and also if username not found

**Review by andymarke — APPROVED** [review]
lgtm
